pinot-core/src/test/java/org/apache/pinot/queries/TextSearchQueriesTest.java

```diff
@@ -138,27 +146,41 @@ protected List<IndexSegment> getIndexSegments() {
   public void setUp()
       throws Exception {
     FileUtils.deleteQuietly(INDEX_DIR);
-
     buildSegment();
-    IndexLoadingConfig indexLoadingConfig = new IndexLoadingConfig();
-    Set<String> textIndexColumns = new HashSet<>();
-    textIndexColumns.addAll(RAW_TEXT_INDEX_COLUMNS);
-    textIndexColumns.addAll(DICT_TEXT_INDEX_COLUMNS);
-    indexLoadingConfig.setTextIndexColumns(textIndexColumns);
-    indexLoadingConfig.setInvertedIndexColumns(new HashSet<>(DICT_TEXT_INDEX_COLUMNS));
-    Map<String, Map<String, String>> columnProperties = new HashMap<>();
+    List<FieldConfig> fieldConfigs = new ArrayList<>();
     Map<String, String> props = new HashMap<>();
     props.put(FieldConfig.TEXT_INDEX_USE_AND_FOR_MULTI_TERM_QUERIES, "true");
-    columnProperties.put(SKILLS_TEXT_COL_MULTI_TERM_NAME, props);
+    fieldConfigs.add(new FieldConfig(SKILLS_TEXT_COL_MULTI_TERM_NAME, FieldConfig.EncodingType.DICTIONARY,
+        FieldConfig.IndexType.TEXT, null, props));
+    fieldConfigs.add(
+        new FieldConfig(QUERY_LOG_TEXT_COL_NAME, FieldConfig.EncodingType.DICTIONARY, FieldConfig.IndexType.TEXT, null,
+            null));
     props = new HashMap<>();
     props.put(FieldConfig.TEXT_INDEX_STOP_WORD_INCLUDE_KEY, "coordinator");
     props.put(FieldConfig.TEXT_INDEX_STOP_WORD_EXCLUDE_KEY, "it, those");
     props.put(FieldConfig.TEXT_INDEX_ENABLE_PREFIX_SUFFIX_PHRASE_QUERIES, "true");
-    columnProperties.put(SKILLS_TEXT_COL_NAME, props);
+    fieldConfigs.add(
+        new FieldConfig(SKILLS_TEXT_COL_NAME, FieldConfig.EncodingType.DICTIONARY, FieldConfig.IndexType.TEXT, null,
+            props));
+    fieldConfigs.add(
+        new FieldConfig(SKILLS_TEXT_NO_RAW_NAME, FieldConfig.EncodingType.DICTIONARY, FieldConfig.IndexType.TEXT, null,
+            null));
+    fieldConfigs.add(
+        new FieldConfig(SKILLS_TEXT_MV_COL_NAME, FieldConfig.EncodingType.DICTIONARY, FieldConfig.IndexType.TEXT, null,
+            null));
     props = new HashMap<>();
     props.put(FieldConfig.TEXT_INDEX_STOP_WORD_EXCLUDE_KEY, "");
-    columnProperties.put(SKILLS_TEXT_COL_DICT_NAME, props);
-    indexLoadingConfig.setColumnProperties(columnProperties);
+    fieldConfigs.add(
+        new FieldConfig(SKILLS_TEXT_COL_DICT_NAME, FieldConfig.EncodingType.DICTIONARY, FieldConfig.IndexType.TEXT,
+            null, props));
+    fieldConfigs.add(
+        new FieldConfig(SKILLS_TEXT_MV_COL_DICT_NAME, FieldConfig.EncodingType.DICTIONARY, FieldConfig.IndexType.TEXT,
+            null, null));
+
+    _tableConfig = new TableConfigBuilder(TableType.OFFLINE).setTableName(TABLE_NAME)
```

Comment:
Seems we are not reusing table config here. Do we need to construct a new one from scratch?

Reply:
In buildSegment() we created one which doesn't include any column properties for field configs to be used in SegmentGeneratorConfig, so tried to create a new one for this with those details included.